CRM/Core/DAO.php

```diff
@@ -1665,9 +1665,7 @@ public static function &copyGeneric($daoName, $criteria, $newData = NULL, $field
         }
 
         if ($newData) {
-          foreach ($newData as $k => $v) {
-            $newObject->$k = $v;
-          }
+          $newObject->copyValues($newData);
```

Comment:
@pradpnayak this is kinda deep & replaces a simple line with a complex one - what is it achieving?

Reply:
1. CopyValues method is more refined in assigning values to DAO object eg incase of '' with 'null'. 
2. $newData is only set if we want to over-ride the event template values in this case when creating event from UI by using template and user changing few fields before hitting save button. 
3. $newData may have empty values for certain parameters which i think DAO ends up converting them into 0 before saving them in db which results in fatal error incase of Foreign Key Constraints.

Reply:
This is the biggest change as it impacts anywhere that uses the copyGeneric function.  However, I've run it through xdebug and it seems to do the right thing - so is likely to fix any obscure bugs anywhere else copyGeneric is used too.